docs/security-assessment.md

```diff
@@ -0,0 +1,99 @@
+| ------------------------- | ------------------------- |
+|1 | Vue User Interface (Frontend) |
+|2 | Stock View Controller |
+|3 |	H2 Database |
+|4 |	Data Response Controller |
+|5 |	Data Request Controller |
+
+## Dataflow Diagram
+To be added by @szymonkowalcyk
+
+## Vulnerabilities & Threats
+| V001 | Lack of authentication & authorization mechanisms |
+| ------------------------- | ------------------------- |
+| Element | PURIS IAM Policy |
+| Before Mitigation | Impact: High, Likelihood: High, Risk: High |
+| After Mitigation | Impact: Low, Likelihood: Low, Risk: Low |
+| Mitigation | Authentication & authorization concept implemented. The front-end provides a Key Cloak integration. (C-X Central Identity Provider is in use.) Puris product may be accessed only by authenticated and authorized personnel, it restricts accessible views based on the client roles: PURIS_USER - common views related to short term information needs PURIS_ADMIN - EDC related views (may be used for debugging)|
+
+| V002 | Lack of User session management controls, including token handling. That pose a security risk, that can lead to unauthorized access and session hijacking. |
+| ------------------------- | ------------------------- |
+| Element | PURIS Product |
+| Before Mitigation | Impact: High, Likelihood: High, Risk: High |
+| After Mitigation | Impact: Low, Likelihood: Low, Risk: Low |
+| Mitigation | Local implementation of KeyCloak was done. Token renewals are enforced for all Users. Token renewal was setup for 5 minutes. |
+
+| V003 | Lack of Input Validation mechanisms implemented, that can lead to various security vulnerabilities related to injection attacks. |
+| ------------------------- | ------------------------- |
+| Element | PURIS Product |
+| Before Mitigation | Impact: High, Likelihood: High, Risk: High |
+| After Mitigation | Impact: Low, Likelihood: Low, Risk: Low |
+| Mitigation | Business Wise Validation was implemented. Input validation on controller level was implemented. Pattern validation on all fields and on entity level was implemented. |
+
+| V004 | Lack of encryption for data in transit, that may expose sensitive data to the risk of interception and compromise.	 |
+| ------------------------- | ------------------------- |
+| Element | PURIS Product |
+| Before Mitigation | Impact: High, Likelihood: High, Risk: High |
+| After Mitigation | Impact: Low, Likelihood: Low, Risk: Low |
+| Mitigation | Product Team currently working on SSL. Team already implemented and enforce HTTPS for  for the front end of the product.Additionally tom is able to serve the backend with spring configuration respectively.Guide was created on how to configure HTTPS  with docker. There are currently some Cores issues that were detected and are being investigated, probably needs spring reconfiguration. It was fixed. Issues happened cause of self signed certificates. The issue is already solved. SSL was also already integrated to the Product. |
```

Comment:
product.Additionally & respectively.Guide -> please add blank.

for  for -> double word and blanks

HTTPS    with docker -> double blanks

"Additionally tom is able to serve" -> Additionally, the admin guide explains how to serve the backend with spring configuration. It also includes configuration of HTTPS with docker.

Reply:
Done. Thanks!